### ISSUE TITLE
There was an error:"PermissionError: [WinError 32] "after I ran it.

### ISSUE THREAD

**Issue by tangyifann** [matthiasdemuzere/w2w#52]
The specific errors  are as follows：
  File "C:\Users\Programs\Python\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Programs\Python\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:\pycharm\venv\Scripts\w2w.exe\__main__.py", line 7, in <module>
  File "D:\pycharm\venv\lib\site-packages\w2w\w2w.py", line 145, in main
    nbui_max = add_urb_params_to_wrf(
  File "D:\pycharm\venv\lib\site-packages\w2w\w2w.py", line 946, in add_urb_params_to_wrf
    os.remove(info['dst_lcz_params_file'])
Could you please tell me how to solve the problem?

**Comment by theendlessriver13** [comment]
Hi,
please provide the complete Traceback you encountered and the exact steps to reproduce it (at least on your machine).
- where is your data stored? what exact path?
- was the file opened by another program while running w2w?

**Comment by tangyifann** [comment]
![20220324102916](https://user-images.githubusercontent.com/101397410/159830652-6e3bc96c-d0e7-4364-b756-f46aa0980cce.png)

**Comment by theendlessriver13** [comment]
I tried reproducing this with windows, but I am unable to. Our example runs fine for me...
Can you try running the provided example with the files included in this repo?: https://github.com/matthiasdemuzere/w2w#run-the-tool
I am not sure about the permissions on your system and I'm also no expert on permissions on windows. I can only guess. Your virtual environment seems to be on a different drive as your data, maybe try having them within the same folder and make sure the file is not opened within a netcdf viewer, GIS, other python or similar.

**Comment by hga619** [comment]
I would like to ask if you have solved this problem. The same problem occurred when I used the author's sample data。

**Comment by b-kode** [comment]
Hi all,

As I am also not on windows, I can not test this behaviour myself.
But as @theendlessriver13 is indicating, this behaviour would typically occur if the file is opened by another application (a netcdf viewer, GIS, other python or similar ..).

Can you confirm this is not the case?

@theendlessriver13: would it help if we `.close()` the file explicitly first?

**Comment by matthiasdemuzere** [comment]
Hi all,

As I am also not on windows, I can not test this behaviour myself.
But as @theendlessriver13 is indicating, this behaviour would typically occur if the file is opened by another application (a netcdf viewer, GIS, other python or similar ..).

Can you confirm this is not the case?

@theendlessriver13: would it help if we .close() the file explicitly first?

**Comment by theendlessriver13** [comment]
@tangyifann I still can't reproduce it. It has to be specific to your system settings and drive permissions ?!

```console
PS C:\Users\jonas\Desktop\tmp> virtualenv venv -ppy38
created virtual environment CPython3.8.8.final.0-64 in 436ms
  creator CPython3Windows(dest=C:\Users\jonas\Desktop\tmp\venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=C:\Users\jonas\AppData\Local\pypa\virtualenv)
    added seed packages: pip==21.0.1, setuptools==60.9.3, wheel==0.37.1
  activators BashActivator,BatchActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
PS C:\Users\jonas\Desktop\tmp> .\venv\Scripts\activate
(venv) PS C:\Users\jonas\Desktop\tmp> git clone git@github.com:matthiasdemuzere/w2w
Cloning into 'w2w'...
remote: Enumerating objects: 978, done.
remote: Counting objects: 100% (978/978), done.
remote: Compressing objects: 100% (593/593), done.
remote: Total 978 (delta 585), reused 704 (delta 364), pack-reused 0
Receiving objects: 100% (978/978), 19.29 MiB | 7.47 MiB/s, done.
Resolving deltas: 100% (585/585), done.
(venv) PS C:\Users\jonas\Desktop\tmp> cd .\w2w\
(venv) PS C:\Users\jonas\Desktop\tmp\w2w> pip install -q .
(venv) PS C:\Users\jonas\Desktop\tmp\w2w> w2w ./sample_data lcz_zaragoza.tif geo_em.d04.nc
--> Set data, arguments and files
> Using layer 0 of the LCZ GeoTIFF. Can be changed with -l (--lcz-band)
--> Check LCZ integrity, in terms of class labels, projection and extent
> LCZ labels as expected (1 to 17)
> LCZ provided as WGS84 (EPSG:4326)
> LCZ domain is covering WRF domain
--> Replace WRF MODIS urban LC with surrounding natural LC
--> Create temporary WRF grid .tif file for resampling
--> Create LCZ-based geo_em file
> Processing LP_URB2D ...
> Processing MH_URB2D ...
> Processing STDH_URB2D ...
> Processing HGT_URB2D ...
> Processing LB_URB2D ...
> Processing LF_URB2D ...
> Processing HI_URB2D ...
--> Create LCZ-based urban extent geo_em file (excluding other LCZ-based info)
--> Expanding land categories of parent domain(s) to 41
> Parent domain d01.nc already contains 41 LC classes
> Parent domain d02.nc already contains 41 LC classes
> Parent domain d03.nc already contains 41 LC classes

--> Start sanity check and clean-up ...
> Check 1: Urban class removed from sample_data\geo_em.d04_NoUrban.nc? OK
> Check 2: LCZ Urban extent present in sample_data\geo_em.d04_LCZ_extent.nc? OK
> Check 3: Urban LCZ classes exists in sample_data\geo_em.d04_LCZ_params.nc? OK: LCZ Classes ([2, 3, 5, 6, 8, 9]) present
> Check 4: FRC_URB2D present in sample_data\geo_em.d04_LCZ_params.nc? OK: FRC_URB2D values range between 0.00 and 0.90
> Check 5: URB_PARAMS matrix present in file sample_data\geo_em.d04_LCZ_params.nc? OK
> Check 6: Do URB_PARAM variable values follow expected range in sample_data\geo_em.d04_LCZ_params.nc?
   + OK for LP_URB2D
   + OK for MH_URB2D
   + OK for STDH_URB2D
   + OK for HGT_URB2D
   + OK for LB_URB2D
   + OK for LF_URB2D
> Check 7: Does HI_URB2D sum to 100% for urban pixels in sample_data\geo_em.d04_LCZ_params.nc? OK
> Check 8: Do FRC_URB and LCZs (from LU_INDEX) cover same extent in sample_data\geo_em.d04_LCZ_params.nc? OK
> Check 9: Extent and # urban pixels same for *_extent.nc and *_params.nc output file? OK, urban extent the same (369)
> Cleaning up ... all done!


 ----------- !! NOTE !! ---------
 Set nbui_max to 5 during compilation, in order to optimize memory storage.
```

@matthiasdemuzere this should all be handled by the packages...

**Comment by tangyifann** [comment]
The same problem also occurred when I used the sample data,and the file wasn't opened by another application. @theendlessriver13 @hga619

**Comment by tangyifann** [comment]
I will reproduce your steps with windows ,thank you ! Hope I can make it. @theendlessriver13

**Comment by hga619** [comment]
Hi，
Have you solved the problem yet.

**Comment by theendlessriver13** [comment]
Unfortunately we can't solve a problem we can't reproduce on our end.
Can you at least confirm that the issue remains with the latest version? Upgrade it with `pip install -U w2w`and make sure `w2w --version` shows `0.2.0`.

**Comment by hga619** [comment]
I just found that my version is not 0.2.0, now it is 0.2.0, I succeed. Thank you so much. @theendlessriver13

**Comment by theendlessriver13** [comment]
Okay, great! Did you install `0.2.0` from Pypi (pip install -U w2w`) or from the `use-tmpfile` branch on github, as I described above. If the latter, can you also try the version from pypi, then we can be sure, the bug was already fixed and further changes are not strictly necessary.

**Comment by hga619** [comment]
I installed 0.2.0 from Pypi.

**Comment by theendlessriver13** [comment]
Okay, thanks for reporting back. Closing this as it is fixed with `0.2.0`.
